hashring/hashring.go

```diff
@@ -66,53 +69,46 @@ func (r *HashRing) RegisterListener(l events.EventListener) {
 	r.listeners = append(r.listeners, l)
 }
 
-func NewHashRing(hashfunc func([]byte) uint32, replicaPoints int) *HashRing {
+// New instantiates and returns a new HashRing,
+func New(hashfunc func([]byte) uint32, replicaPoints int) *HashRing {
 	ring := &HashRing{
-		hashfunc:      hashfunc,
 		replicaPoints: replicaPoints,
+		hashfunc: func(str string) int {
+			return int(hashfunc([]byte(str)))
+		},
 	}
 
 	ring.servers.byAddress = make(map[string]bool)
-	ring.servers.tree = &RBTree{}
-
+	ring.servers.tree = &RedBlackTree{}
 	return ring
 }
 
+// Checksum returns the checksum of all stored servers in the HashRing
+// Use this value to find out if the HashRing is mutated.
 func (r *HashRing) Checksum() uint32 {
 	r.servers.RLock()
-	checksum := r.servers.checksum
-	r.servers.RUnlock()
-
-	return checksum
+	defer r.servers.RUnlock()
```

Comment:
Is there a reason you rewrote the function to use `defer`? `defer` has a [non-negligible overhead](http://lk4d4.darth.io/posts/defer/)